lib/BuildSystem/BuildExecutionQueue.cpp

```diff
@@ -32,7 +33,16 @@ BuildExecutionQueue::~BuildExecutionQueue() {
 
 CommandResult BuildExecutionQueue::executeProcess(
     QueueJobContext* context, ArrayRef<StringRef> commandLine) {
-  return executeProcess(context, commandLine, {});
+  // Promises are move constructible only, thus cannot be put into std::function
+  // objects that themselves get copied around. So we must create a shared_ptr
+  // here to allow it to go along with the labmda.
+  std::shared_ptr<std::promise<CommandResult>> p{new std::promise<CommandResult>};
```

Comment:
You can std::move it in to the lambda, but of course you then need to std::move the lambda around too.  I'm sure this could be done, though it complicates the common case.